frontend/ui/lib/model/place_visit.dart

```diff
@@ -8,9 +8,19 @@ enum UserMark {
 }
 
 class PlaceVisit {
+  static final Map<String, UserMark> stringToUserMark = {
```

Comment:
The problem with this approach is that it's hard to make sure that this stays in sync with the enum values themselves. This should instead create a map directly from the enum values themselves, e.g.:

```dart
  final Map<String, UserMark> stringToUserMark = Map.fromIterable(
    UserMark.values,
    key: (v) {
      final s = v.toString();
      final dotIndex = s.lastIndexOf(".");
      return s.substring(dotIndex + 1);
    },
    value: (v) => v,
  );
```

Feel free to make this change in a follow-up PR.